mapfishapp/src/main/webapp/app/js/GEOR_Querier.js

```diff
@@ -0,0 +1,320 @@
+            layer = r.get('layer'), 
+            name = r.get('title') || layer.name || '';
+
+        this.title = OpenLayers.i18n("Request on NAME", {
+            'NAME': name
+        });
+
+        this.pseudoRecord = {
+            typeName: isWFS ? 
+                r.get("WFS_typeName") : r.get("name"),
+            owsURL: isWFS ? 
+                layer.protocol.url : r.get("WFS_URL")
+        };
+
+        GEOR.waiter.show();
+        // get layer model through WFS DescribeFeatureType:
+        this.attributeStore = GEOR.ows.WFSDescribeFeatureType(this.pseudoRecord, {
+            extractFeatureNS: true,
+            success: function() {
+                // we list all fields, including the geometry
+                this.layerFields = this.attributeStore.collect('name');
+                // we get the geometry column name, and remove the corresponding record from store
+                var idx = this.attributeStore.find('type', GEOR.ows.matchGeomProperty);
+                if (idx > -1) {
+                    // we have a geometry
+                    var r = this.attributeStore.getAt(idx);
+                    this.geometryName = r.get('name');
+                    this.attributeStore.remove(r);
+                } else {
+                    GEOR.util.infoDialog({
+                        msg: tr("querier.layer.no.geom")
+                    });
+                }
+            },
+            failure: function() {
+                GEOR.util.errorDialog({
+                    msg: tr("querier.layer.error")
+                });
+            },
+            scope: this
+        });
+
+        this.layer = new OpenLayers.Layer.Vector('__georchestra_filterbuilder', {
+            displayInLayerSwitcher: false,
+            styleMap: GEOR.util.getStyleMap({
+                "default": {
+                    strokeWidth: 2,
+                    strokeColor: "#ee5400",
+                    fillOpacity: 0
+                }
+            })
+        });
+
+        this.filterbuilder = new Styler.FilterBuilder({
+            defaultBuilderType: Styler.FilterBuilder.ALL_OF,
+            filterPanelOptions: {
+                attributesComboConfig: {
+                    displayField: "name",
+                    listWidth: 165,
+                    tpl: GEOR.util.getAttributesComboTpl()
+                }
+            },
+            attributes: this.attributeStore,
+            toolbarType: "tbar",
+            map: this.map,
+            allowGroups: true,
+            noConditionOnInit: false,
+            deactivable: true,
+            autoScroll: true,
+            allowSpatial: true,
+            cookieProvider: this.cookieProvider,
+            vectorLayer: this.layer
+        });
+
+        this.items = [this.filterbuilder];
+        this.buttons = [{
+            text: OpenLayers.i18n("Close"),
+            handler: this.close,
+            scope: this
+        }, {
+            text: OpenLayers.i18n("Search"),
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ["OpenLayers" is not defined.](https://www.codacy.com/app/georchestra/georchestra/file/2337677315/issues/source?bid=3200181&fileBranchId=3209168#l187)